gateway/sensor.go

```diff
@@ -106,9 +106,19 @@ func NewGateway(
 		Named:   conf.ResourceName().AsNamed(),
 		logger:  logger,
 		started: false,
+		workers: utils.NewBackgroundStoppableWorkers(),
+	}
+
+	// Need to disable buffering on stdout so C logs can be displayed in real time.
+	C.disableBuffering()
+
+	// capture c log output
+	err := g.captureOutput()
```

Comment:
[opt] to go along with the other comment, I think adding to the worker from the NewGateway function makes its clear to a reader that a background thread is being spun up. note you would technically need to add a context as an input to captureOutput

```suggestion
		started: false,
	}

	// Need to disable buffering on stdout so C logs can be displayed in real time.
	C.disableBuffering()

	// capture c log output
	g.workers =  utils.NewBackgroundStoppableWorkers(g.captureOutput)
```